extensions/data-workspace/src/main.ts

```diff
@@ -16,10 +16,14 @@ import { ProjectDashboard } from './dialogs/projectDashboard';
 
 export async function activate(context: vscode.ExtensionContext): Promise<IExtension> {
 	const workspaceService = new WorkspaceService(context);
-	await workspaceService.loadTempProjects();
-	await workspaceService.checkForProjectsNotAddedToWorkspace();
-	context.subscriptions.push(vscode.workspace.onDidChangeWorkspaceFolders(async () => {
-		await workspaceService.checkForProjectsNotAddedToWorkspace();
+
+	// this is not being awaited to not block the rest of activate function from running while loading any temp projects and
+	// checking for projects not added to the workspace yet
+	workspaceService.loadTempProjects().then(async () => await workspaceService.checkForProjectsNotAddedToWorkspace())
+		.catch(e => console.log(e));
```

Comment:
1. I see a mixed use of await and then, not a big deal, but I think you can do 
```
workspaceService.loadTempProjects().then(() => { 
return workspaceService.checkForProjectsNotAddedToWorkspace();
}). catch(e => console.log(e));
```
2. do you really not want to notify the user about the failure here? IIUC, it does have a user impact.

Reply:
good point. It's better to have an error show as an error, rather than just log. Should I also add a notification about an error, or is it enough to just have it in the console since it wouldn't be common to have an error from this scenario?

Reply:
I think displaying an error to the user in this case is warranted so they know something went wrong and might be able to fix it.